File: Code/new_fitervari_backend/src/main/java/at/htl/control/PersonRepository.java
package at.htl.control;

import at.htl.model.Person;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;
import javax.transaction.Transactional;

@ApplicationScoped
public class PersonRepository implements PanacheRepository<Person> {

    public Person save(Person entity) {
        return getEntityManager().merge(entity);
    }

    public boolean checkIfCustomerAlreadyExists(String name) {
        return streamAll().anyMatch(person -> person.getKeycloakName().equals(name));
    }

    public Person getPersonByName(String name) {
        return streamAll()
                .filter(person -> person.getKeycloakName().equals(name))
                .findFirst().get();
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/PersonRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.NewsLetter
import at.htl_leonding.model.Person
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class PersonRepository : PanacheRepository<Person> {
    @Inject
    lateinit var em: EntityManager

    fun save(person: Person): Person {
        return em.merge(person)
    }
}